core/templates/services/prevent-page-unload-event.service.spec.ts

```diff
@@ -101,4 +103,31 @@ describe ('Prevent page unload event service', function() {
 
     expect(preventPageUnloadEventService.removeListener).toHaveBeenCalled();
   });
+
+  it('should test if Alert is displayed when a condition is passed', () => {
+    spyOnProperty(windowRef, 'nativeWindow', 'get').and.returnValue(mockWindow);
+    preventPageUnloadEventService.addListener(() => {
+      return true;
+    });
+    spyOn(reloadEvt, 'preventDefault');
+
+    windowRef.nativeWindow.location.reload();
+
+    expect(reloadEvt.preventDefault).toHaveBeenCalled();
+    expect(preventPageUnloadEventService.isListenerActive()).toBe(true);
+  });
+
+  it('should test if Alert is not displayed when a condition is passed', () => {
+    spyOnProperty(windowRef, 'nativeWindow', 'get').and.returnValue(mockWindow);
+    var val = () => {
```

Comment:
What is `val`? Please give this a more meaningful name.

Reply:
DONE